### ISSUE TITLE
Undefined behavior in guitargraph.cc

### ISSUE THREAD

**Issue by deuteragenie** [performous/performous#602]
Line 629 of guitargraph.cc, neckglow.w could have a garbage value, and the behavior undefined.
This is reported by clang static analyzer.

**Comment by Tronic** [comment]
That line number does not match the file in git master but I suppose you mean this variable here
https://github.com/performous/performous/blob/master/game/guitargraph.cc#L776

The vector is supposed to be default-initialised all zeroes, and it was before, but the PR #390 and more specifically its commit https://github.com/performous/performous/commit/983a6dab1e2a5a9d08a88ad9fb69c1a805e2fd3a replaced the original implementation in glmath.hh with typedefs using an external GLM library instead. The problem is that the default constructors of the GLM library do not initialise the vectors like our own constructors did.

I think the most suitable fix here is to explicitly pass four zeroes on line 776. There are likely to be similar uses elsewhere in Performous where this could be an issue, but hopefully the static analysis was complete and this is in fact the only such occurrence.

**Comment by Baklap4** [comment]
> I think the most suitable fix here is to explicitly pass four zeroes on line 776

That should get the job done yes 👍

**Comment by deuteragenie** [comment]
Yes, it was the only occurrence in the whole tree (there are a few more legacy stuff on the ced files though).

**Comment by Lord-Kamina** [comment]
> That line number does not match the file in git master but I suppose you mean this variable here
> https://github.com/performous/performous/blob/master/game/guitargraph.cc#L776
> 
> The vector is supposed to be default-initialised all zeroes, and it was before, but the PR #390 and more specifically its commit [983a6da](https://github.com/performous/performous/commit/983a6dab1e2a5a9d08a88ad9fb69c1a805e2fd3a) replaced the original implementation in glmath.hh with typedefs using an external GLM library instead. The problem is that the default constructors of the GLM library do not initialise the vectors like our own constructors did.
> 
> I think the most suitable fix here is to explicitly pass four zeroes on line 776. There are likely to be similar uses elsewhere in Performous where this could be an issue, but hopefully the static analysis was complete and this is in fact the only such occurrence.

Correct. GLM's behavior used to be the same as the in-house glmath performous used to have but that was changed in `0.9.9.0`. Not sure if this reflected a change in the GLSL standard or what but specifically initializing to `0.0f` is the solution. Not sure if it's actually required to explicitly specify all components of the vector (matrices work with a single parameter) but it might be clearer to read.

**Comment by deuteragenie** [comment]
The issue is still present on master, line 829.

```
// Mangle neck glow color as needed
--
828 | // Convert sum into average and apply correctness as premultiplied alpha
829 | if (neckglow.w > 0.0) neckglow = (correctness() / neckglow.w) * neckglow;
```
neglow.w is undefined/garbage.

**Comment by Baklap4** [comment]
Is this issue still a thing since https://github.com/performous/performous/pull/737 got merged. The neckglow variable was unitialized, and triggered errors upon debug builds on Windows. 
Please reopen if the latest master still has this issue
